pulsar-broker/src/main/java/org/apache/pulsar/broker/TransactionMetadataStoreService.java

```diff
@@ -319,8 +319,14 @@ public long getLowWaterMark(TxnID txnID) {
         return store.updateTxnStatus(txnId, newStatus, expectedStatus, isTimeout);
     }
 
-    public CompletableFuture<Void> endTransaction(TxnID txnID, int txnAction, boolean isTimeout) {
-        CompletableFuture<Void> completableFuture = new CompletableFuture<>();
+    public CompletableFuture<Void> endTransaction(TxnID txnID, int txnAction, boolean isTimeout,
```

Comment:
```    
public CompletableFuture<Void> endTransaction(TxnID txnID, int txnAction, boolean isTimeout) {
        CompletableFuture<Void> completableFuture = new CompletableFuture<>();
        return endTransaction txnID, txnAction, isTimeout, completableFuture);
}
```
code like this may be better